pkg/pulumiyaml/run.go

```diff
@@ -703,28 +750,31 @@ func (r *Runner) ensureSetup(ctx *pulumi.Context) {
 		"stack":   stack,
 	}
 	r.cwd = cwd
+}
 
-	r.intermediates = []graphNode{}
-
-	// Topologically sort the intermediates based on implicit and explicit dependencies
-	intermediates, rdiags := topologicallySortedResources(r.t)
-	r.sdiags.Extend(rdiags...)
-	if rdiags.HasErrors() {
-		return
+func getConfigType(v string) ctypes.Type {
```

Comment:
This doesn't work for array types. I'll open a PR into your branch that will fix this.